src/Plugin.php

```diff
@@ -3069,7 +3068,9 @@ public static function processMassiveActionsForOneItemtype(
                     $plugin->getFromDB($id);
                     if ($plugin->isInstalled($plugin->fields['directory'])) {
                         $plugin->uninstall($id);
-                        if (!$plugin->isInstalled($plugin->fields['directory'])) {
+                        // Note: phpstan doesn't seems to understand here that $plugin->uninstall will modify the return
+                        // value of $plugin->isInstalled().
+                        if (!$plugin->isInstalled($plugin->fields['directory'])) { // @phpstan-ignore booleanNot.alwaysFalse
```

Comment:
Probably need to mark functions as impure

Reply:
Maybe this method should return a boolean, then we would be able to check its result.

```php
if ($plugin->uninstall($id)) {
    $ma->itemDone($item->getType(), $id, MassiveAction::ACTION_OK);
} else {
    $ma->itemDone($item->getType(), $id, MassiveAction::ACTION_KO);
}
```

It is not supposed to be used by plugins, so we could change its signature.

Reply:
Same remark for activate/unactivate methods.

Reply:
Done 7e415f5d7e.